gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -122,6 +122,19 @@ public TransportChannelProvider withEndpoint(String endpoint) {
     return toBuilder().setEndpoint(endpoint).build();
   }
 
+  @Override
+  @BetaApi("The surface for customizing pool size is not stable yet and may change in the future.")
+  public boolean acceptsPoolSize() {
+    return poolSize == null;
+  }
+
+  @Override
+  @BetaApi("The surface for customizing pool size is not stable yet and may change in the future.")
+  public InstantiatingGrpcChannelProvider withPoolSize(int size) {
```

Comment:
Do you have any preference between returning the interface vs the concrete type?

Reply:
If all the available functionality is accessible through the interface, then we don't need to return the concrete type. All of the other "withX" methods return the interface, so this should be consistent.